libopendavinci/src/odcore/base/module/ManagedClientModule.cpp

```diff
@@ -116,12 +121,106 @@ namespace odcore {
                 return m_startOfLastCycle;
             }
 
+            std::vector<double> ManagedClientModule::computeAdvancedStatistics() {
+#ifdef __linux__
+                bool success=false;
+                double cpu_stats[4];
+                
+                try
+                {
+                    // get system info
+                    
+                    // The number of processors currently online
+                    int numCPU = sysconf(_SC_NPROCESSORS_ONLN);
+                    if(numCPU == -1) OPENDAVINCI_CORE_THROW_EXCEPTION(IOException,"Could not get system info");
+                    // Size of a page in bytes
+                    int pagesize_in_bytes = sysconf(_SC_PAGESIZE);
+                    if(pagesize_in_bytes < 1) OPENDAVINCI_CORE_THROW_EXCEPTION(IOException,"Could not get system info");
+                    // The number of clock ticks per second
+                    long tickspersec = sysconf(_SC_CLK_TCK);
+                    if(tickspersec == -1) OPENDAVINCI_CORE_THROW_EXCEPTION(IOException,"Could not get system info");
+                    
+                    double utime, stime, start_time, total_vm_size, resident_set_size;
+                    
+                    ifstream proc_file;
+                    proc_file.open("/proc/self/stat");
+                    {
+                        uint64_t temp_buffer=0;
+                        char chr;
+                        string str;
+                        // discard values of no interest
+                        proc_file>>temp_buffer;
+                        proc_file>>str;
+                        proc_file>>chr;
+                        for(uint8_t i=0;i<11;++i)
+                            proc_file>>temp_buffer;
+                        // Amount of time that this process has been scheduled in user mode (in clock ticks)
+                        utime=(double)temp_buffer/tickspersec/numCPU;
+                        proc_file>>temp_buffer;
+                        // Amount of time that this process has been scheduled in kernel mode (in clock ticks)
+                        stime=(double)temp_buffer/tickspersec/numCPU;
+                        // discard values of no interest
+                        for(uint8_t i=0;i<7;++i)
+                            proc_file>>temp_buffer;
+                        // The time the process started after system boot
+                        start_time=(double)temp_buffer/tickspersec;
+                        // Virtual memory size in bytes
+                        proc_file>>temp_buffer;
+                        total_vm_size=temp_buffer;
```

Comment:
They are doubles because they are assigned to the same array together with floating point % variables.
Integer (re-)conversion is performed at a later step.